bash-code-style.md

```diff
@@ -0,0 +1,161 @@
+# Writing Bash Scripts
+
+## When to use bash
+
+If you are coordinating a bunch of other utilities and it is not a complex
+process bash is a good choice.  It can also be helpful for prototyping quickly
+before reaching for a more powerful language.  As a rule of thumb bash scripts
+should never grow larger than a few hundred lines.  You probably want a better
+language at that point.
```

Comment:
Maybe a bit too opinionated, maybe  "You probably want a better language at that point." could be rephrased to: "At this stage you might want to re-evaluate choosing bash"